### PR TITLE
[testing] Add multi-core support to TCP echo

### PR DESCRIPTION
**Pull request by iyzhang**
This PR add support for multicore execution to our tcp-echo benchmark as an example of how we will handle basic multi-core architectures with RSS support. It currently only works with Catnap because there is more configuration that needs to be done for DPDK support. Since it has a share nothing architecture, it should work fine with Catloop and Catmem but is not currently supported.

### CODE REVIEW

File: src/rust/catnap/linux/transport.rs [hunk 1]
```diff
@@ -294,6 +294,7 @@ impl NetworkTransport for SharedCatnapTransport {
                     error!("new(): {}", cause);
                     return Err(Fail::new(get_libc_err(e), &cause));
                 }
+
```
Comment (ppenna) [inline src/rust/catnap/linux/transport.rs:297]: Could you evert this line?

File: examples/tcp-echo/main.rs [hunk 2]
```diff
@@ -110,7 +114,15 @@ impl ProgramArguments {
...
-                    .help("Sets number of clients"),
+                    .help("Sets number of clients (per thread)"),
+            )
+            .arg(
+                Arg::new("nthreads")
+                    .long("nthreads")
+                    .value_parser(clap::value_parser!(usize))
+                    .required(false)
+                    .value_name("NUMBER")
+                    .help("Sets number of threads"),
```
Comment (ppenna) [inline examples/tcp-echo/main.rs:125]: This option should be available only on targets that support this.

File: src/rust/demikernel/libos/name.rs [hunk 3]
```diff
@@ -12,6 +12,7 @@ use ::std::env;
 // Structures
 //======================================================================================================================
 
+#[derive(Clone, Copy)]
```
Comment (ppenna) [inline src/rust/demikernel/libos/name.rs:15]: Why is this change required?

File: tools/ci/config/ci_map.yaml [hunk 4]
```diff
@@ -121,6 +121,18 @@ catnap:
...
+      nclients: 32
+      nrequests: 1024
+      run_mode: concurrent
+      nthreads: 2
+    scenario9:
+      bufsize: 64
+      nclients: 32
+      nrequests: 1024
+      run_mode: concurrent
+      nthreads: 4
```
Comment (ppenna) [inline tools/ci/config/ci_map.yaml:135]: This requires the CI tool to be updated.

File: tools/ci/config/ci_map.yaml
Comment (ppenna) [inline tools/ci/config/ci_map.yaml:1]: Something else is required to make this change work out of the box for the other LibOSes.

Comment (iyzhang) [reply tools/ci/config/ci_map.yaml:1]: Hmm, I thought I turned it off for all other libOSes...
